L1Trigger/L1TTrackMatch/plugins/L1TrackSelectionProducer.cc

```diff
@@ -255,6 +276,66 @@ class L1TrackSelectionProducer : public edm::global::EDProducer<> {
   private:
     double reducedChi2RPhiMax_;
   };
+  struct TTTrackChi2MaxNstubSelector {
+    TTTrackChi2MaxNstubSelector(double reducedChi2MaxNstub4, double reducedChi2MaxNstub5)
+        : reducedChi2MaxNstub4_(reducedChi2MaxNstub4), reducedChi2MaxNstub5_(reducedChi2MaxNstub5) {}
+    TTTrackChi2MaxNstubSelector(const edm::ParameterSet& cfg)
+        : reducedChi2MaxNstub4_(cfg.template getParameter<double>("reducedChi2MaxNstub4")),
+          reducedChi2MaxNstub5_(cfg.template getParameter<double>("reducedChi2MaxNstub5")) {}
+    bool operator()(const L1Track& t) const {
+      return (((t.chi2Red() < reducedChi2MaxNstub4_) && (t.getStubRefs().size() == 4)) ||
+              ((t.chi2Red() < reducedChi2MaxNstub5_) && (t.getStubRefs().size() > 4)));
+    }
+
+  private:
+    double reducedChi2MaxNstub4_;
+    double reducedChi2MaxNstub5_;
+  };
+  struct TTTrackWordChi2MaxNstubSelector {  // using simulated chi2 since not implemented in track word, updates needed
+    TTTrackWordChi2MaxNstubSelector(double reducedChi2MaxNstub4, double reducedChi2MaxNstub5)
+        : reducedChi2MaxNstub4_(reducedChi2MaxNstub4), reducedChi2MaxNstub5_(reducedChi2MaxNstub5) {}
+    TTTrackWordChi2MaxNstubSelector(const edm::ParameterSet& cfg)
+        : reducedChi2MaxNstub4_(cfg.template getParameter<double>("reducedChi2MaxNstub4")),
+          reducedChi2MaxNstub5_(cfg.template getParameter<double>("reducedChi2MaxNstub5")) {}
+    bool operator()(const L1Track& t) const {
+      return (((t.chi2Red() < reducedChi2MaxNstub4_) && (t.getNStubs() == 4)) ||
+              ((t.chi2Red() < reducedChi2MaxNstub5_) && (t.getNStubs() > 4)));
```

Comment:
Presumably these selectors duplicate what was in TrackJets. It looks like chi2Red() is not a TrackWord method, however [getNStubs(), getBendChi2() accessing just TrackWord info]. Since we're moving towards selecting on MVAs (eventually binned and in the TrackWord), this shouldn't matter, but we are temporarily breaking from these selectors exclusively using TrackWord info and we won't be able to match in Firmware.

Reply:
Right, this is why I have a note that this needs to be updated. Although we have swapped to MVA for track jets, the displaced track jets still use chi2 and that why this is here but it's meant to be temporary. This necessary update will be brought up to the GTT group this week.